app/models/registration/policy.rb

```diff
@@ -16,5 +16,51 @@ class Policy < ApplicationRecord
     validates :position, uniqueness: { scope: :registration_campaign_id }
 
     acts_as_list scope: :registration_campaign
+
+    scope :active, -> { where(active: true) }
+
+    scope :for_phase, lambda { |phase|
+      where(phase: [phases[:both], phases[phase]])
+    }
+
+    def evaluate(user)
+      case kind.to_sym
+      when :institutional_email
+        evaluate_institutional_email(user)
+      when :prerequisite_campaign
+        evaluate_prerequisite_campaign(user)
+      else
+        { pass: true, code: :ok }
+      end
+    end
+
+    private
+
+      def evaluate_institutional_email(user)
+        domains = Array(config&.fetch("allowed_domains", nil)).map do |domain|
+          (domain || "").strip.downcase
+        end.reject(&:empty?)
+
+        return { pass: true, code: :ok } if domains.empty?
+
+        email = user.email.to_s.downcase
+        allowed = domains.any? do |domain|
+          email.end_with?("@#{domain}")
+        end
+
+        { pass: allowed, code: allowed ? :ok : :institutional_email_mismatch }
+      end
+
+      def evaluate_prerequisite_campaign(user)
+        campaign_id = config&.fetch("prerequisite_campaign_id", nil)
+        return { pass: true, code: :ok } if campaign_id.blank?
```

Comment:
Whenever the config cannot be correctly loaded, this indicates a serious flaw and we shouldn't just pass the policy. But not sure if throwing an error would be better here.

Reply:
I handled this differently in the rewrite.